vars/tmsUpload.groovy

```diff
@@ -144,12 +144,19 @@ void call(Map parameters = [:]) {
                 // so that user can get them in one pipeline run
                 // put the validation here, because we need uri and token to call tms get nodes api
                 List nodes = tms.getNodes(uri, token).getAt("nodes");
-                Map nodeIdExtDesMap = validateNodeExtDescriptorMapping(nodeExtDescriptorMapping, nodes, mtaVersion)
+                Map mtaYaml = getMtaYaml();
+                Map nodeIdExtDesMap = validateNodeExtDescriptorMapping(nodeExtDescriptorMapping, nodes, mtaYaml, mtaVersion)
                 
                 if(nodeIdExtDesMap) {
                     nodeIdExtDesMap.each{ key, value ->
-                        def uploadMtaExtDescriptorToNodeResponse = tms.uploadMtaExtDescriptorToNode(uri, token, key, "${workspace}/${value}", mtaVersion, description, namedUser)
-                        echo "[TransportManagementService] MTA Extention Descriptor '${uploadMtaExtDescriptorToNodeResponse.fileName}' (fileId: '${uploadMtaExtDescriptorToNodeResponse.fileId}') successfully uploaded to Node with id '${key}'."
+                        List mtaExtDescriptor = tms.getAMtaExtDescriptor(uri, token, key, mtaYaml.ID, mtaVersion)
```

Comment:
As far as I understand, there will be only a single extension descriptor identified by provided touple (nodeId, mtaId, mtaVersion). Probably, `getAMtaExteDescriptor(...)` method could already return that single one.

Reply:
> As far as I understand, there will be only single extension descriptor identified by provided tuple (nodeId, mtaId, mtaVersion). Probably, `getAMtaExtDescriptor(...)` method could already return that single one.

The API will always return a list no matter with multiple-, single- or empty object inside. I had this concern as well and googled the reason, it seems like to keep the consistency so that it always returns a certain and well-formatted response.

Reply:
To return a list is okay for the API, but we might think to implement the logic in client method in a way that it always takes only single result from the list (we actually expect only a single one). See my comment above.

Reply:
> To return a list is okay for the API, but we might think to implement the logic in client method in a way that it always takes only single result from the list (we actually expect only a single one). See my comment above.

Hi Artem, I got your idea but I just think if we do this we will lose the smart design by API, ie. return a **certain** object (even though not find an element, still return 200 with empty list  rather than 404). That means when we extract the object from the list in the client method, we need to check if the list is empty or not. If empty then returns an empty Map(or a null object), if not, return a map (or other kind of object). And then in the tmsUpload.groovy, we need to check again whether the object is null or empty, and then do what we want. From my point of view, it might not be necessary.